gsetting.py

```diff
@@ -97,20 +95,31 @@ def main():
     state = module.params['state']
     user = module.params['user']
     schemadir = module.params['schemadir']
-    key = module.params['key']
-    value = module.params['value']
+    settings = module.params['settings']
```

Comment:
Thanks in general this looks nice, but can we keep it backwards compatible?

I guess the key/value/settings parameters would all have to be optional, and we'd throw an error if none is given. There does not need to be much special logic to keep key/value working. When key/value are provided, they can simply get added as one item into the `settings` dict, and all the other code can remain as you've implemented it.

Apologies for the massive delay on reviewing this :).

Reply:
No problem! Sure, makes sense to maintain compatibility - I'll make some changes and come back with a commit shortly.

Reply:
I've made those changes, please take a look. I haven't updated the README this time, so there's no mention of the old key/value in there any more but if you want users to know they have a choice I can put an example back in.